Went through the workflow suggested in the header of this PR. 
When doing the step `create your features selecting the Disable ContentDB Update option`, hit following bug.
In the Feature Creation script window I clicked `Run Script`. Nothing at all happened, the window was there without any changes, the `Run Script` button was still clickable. So I clicked it again. To my surprise, two actions in the Activities started to go on. The former activity finished first, with just deprecation warnings. The latter (=higher) actiivty finished second, with some error - see error and screenshot below.
![screen shot 2013-06-13 at 11 02 28](https://f.cloud.github.com/assets/2478303/647763/195ee272-d411-11e2-95b3-73fe5b7e610f.png)

```
/usr/lib64/python2.6/site-packages/mahotas/texture.py:6: DeprecationWarning: Use

from mahotas.features import texture

  ''', DeprecationWarning)
/usr/lib64/python2.6/site-packages/mahotas/zernike.py:6: DeprecationWarning: Use

from mahotas.features import zernike

  ''', DeprecationWarning)
/usr/lib64/python2.6/site-packages/mahotas/tas.py:6: DeprecationWarning: Use

from mahotas.features import tas

  ''', DeprecationWarning)
/usr/lib64/python2.6/site-packages/mahotas/lbp.py:6: DeprecationWarning: Use

from mahotas.features import lbp

  ''', DeprecationWarning)
WARNING:omero.gateway:OptimisticLockException on <class 'omero.gateway.OmeroGatewaySafeCallWrapper'> to <ed326e43-ba3a-4665-85f9-c8de3ce02946omero.api.IUpdate> saveObject((object #0 (::omero::model::TagAnnotation)
{
    _id = object #1 (::omero::RLong)
    {
        _val = 2341
    }
    _details = object #2 (::omero::model::Details)
    {
        _owner = object #3 (::omero::model::Experimenter)
        {
            _id = object #4 (::omero::RLong)
            {
                _val = 3
            }
            _details = <nil>
            _loaded = False
            _version = <nil>
            _groupExperimenterMapSeq = 
            {
            }
            _groupExperimenterMapLoaded = False
            _omeName = <nil>
            _firstName = <nil>
            _middleName = <nil>
            _lastName = <nil>
            _institution = <nil>
            _email = <nil>
            _annotationLinksSeq = 
            {
            }
            _annotationLinksLoaded = False
            _annotationLinksCountPerOwner = 
            {
            }
        }
        _group = object #5 (::omero::model::ExperimenterGroup)
        {
            _id = object #6 (::omero::RLong)
            {
                _val = 7
            }
            _details = object #7 (::omero::model::Details)
            {
                _owner = <nil>
                _group = <nil>
                _creationEvent = <nil>
                _updateEvent = <nil>
                _permissions = object #8 (::omero::model::Permissions)
                {
                    _restrictions = 
                    {
                        [0] = False
                        [1] = True
                        [2] = True
                        [3] = True
                    }
                    _perm1 = -40
                }
                _externalInfo = <nil>
                _call = 
                {
                    key = omero.session.uuid
                    value = 52557c6f-1a6e-448c-8b46-cb9bf8084b90

                    key = omero.client.uuid
                    value = ed326e43-ba3a-4665-85f9-c8de3ce02946
                }
                _event = object #9 (::omero::sys::EventContext)
                {
                    shareId = -1
                    sessionId = 1379
                    sessionUuid = 52557c6f-1a6e-448c-8b46-cb9bf8084b90
                    userId = 3
                    userName = searcher-1
                    groupId = 7
                    groupName = searcher-1
                    isAdmin = False
                    eventId = -1
                    eventType = User
                    memberOfGroups = 
                    {
                        [0] = 3
                        [1] = 1
                        [2] = 7
                    }
                    leaderOfGroups = 
                    {
                        [0] = 7
                    }
                    groupPermissions = object #10 (::omero::model::Permissions)
                    {
                        _restrictions = 
                        {
                        }
                        _perm1 = -40
                    }
                }
            }
            _loaded = True
            _version = object #11 (::omero::RInt)
            {
                _val = 0
            }
            _name = object #12 (::omero::RString)
            {
                _val = searcher-1
            }
            _groupExperimenterMapSeq = 
            {
            }
            _groupExperimenterMapLoaded = False
            _annotationLinksSeq = 
            {
            }
            _annotationLinksLoaded = False
            _annotationLinksCountPerOwner = 
            {
            }
            _description = <nil>
        }
        _creationEvent = object #13 (::omero::model::Event)
        {
            _id = object #14 (::omero::RLong)
            {
                _val = 68679
            }
            _details = <nil>
            _loaded = False
            _status = <nil>
            _time = <nil>
            _experimenter = <nil>
            _experimenterGroup = <nil>
            _type = <nil>
            _containingEvent = <nil>
            _logsSeq = 
            {
            }
            _logsLoaded = False
            _session = <nil>
        }
        _updateEvent = object #15 (::omero::model::Event)
        {
            _id = object #16 (::omero::RLong)
            {
                _val = 68682
            }
            _details = <nil>
            _loaded = False
            _status = <nil>
            _time = <nil>
            _experimenter = <nil>
            _experimenterGroup = <nil>
            _type = <nil>
            _containingEvent = <nil>
            _logsSeq = 
            {
            }
            _logsLoaded = False
            _session = <nil>
        }
        _permissions = object #17 (::omero::model::Permissions)
        {
            _restrictions = 
            {
            }
            _perm1 = -40
        }
        _externalInfo = <nil>
        _call = 
        {
            key = omero.session.uuid
            value = 52557c6f-1a6e-448c-8b46-cb9bf8084b90

            key = omero.client.uuid
            value = ed326e43-ba3a-4665-85f9-c8de3ce02946
        }
        _event = object #18 (::omero::sys::EventContext)
        {
            shareId = -1
            sessionId = 1379
            sessionUuid = 52557c6f-1a6e-448c-8b46-cb9bf8084b90
            userId = 3
            userName = searcher-1
            groupId = 7
            groupName = searcher-1
            isAdmin = False
            eventId = -1
            eventType = User
            memberOfGroups = 
            {
                [0] = 3
                [1] = 1
                [2] = 7
            }
            leaderOfGroups = 
            {
                [0] = 7
            }
            groupPermissions = object #19 (::omero::model::Permissions)
            {
                _restrictions = 
                {
                }
                _perm1 = -40
            }
        }
    }
    _loaded = True
    _version = <nil>
    _ns = object #20 (::omero::RString)
    {
        _val = direct.edu.cmu.cs.compbio.omepslid:all_slf33
    }
    _description = <nil>
    _annotationLinksSeq = 
    {
    }
    _annotationLinksLoaded = False
    _annotationLinksCountPerOwner = 
    {
    }
    _textValue = object #21 (::omero::RString)
    {
        _val = 7_all_slf33_content_db_00000000000000000003.pkl
    }
},), {})
Traceback (most recent call last):
  File "/opt/omero/server/lib/python/omero/gateway/__init__.py", line 3457, in __call__
    return self.f(*args, **kwargs)
  File "/opt/omero/server/lib/python/omero_api_IUpdate_ice.py", line 107, in saveObject
    return _M_omero.api.IUpdate._op_saveObject.invoke(self, ((obj, ), _ctx))
OptimisticLockException: exception ::omero::OptimisticLockException
{
    serverStackTrace = ome.conditions.OptimisticLockException: You are not authorized to change the update event for ome.model.annotations.TagAnnotation:Id_2341 from ome.model.meta.Event:Id_68717 to ome.model.meta.Event:Id_68682
You may need to reload the object before continuing.
    at ome.security.basic.OmeroInterceptor.managedEvent(OmeroInterceptor.java:1003)
    at ome.security.basic.OmeroInterceptor.checkManagedDetails(OmeroInterceptor.java:773)
    at ome.security.basic.OmeroInterceptor.resetDetails(OmeroInterceptor.java:316)
    at ome.security.basic.OmeroInterceptor.onFlushDirty(OmeroInterceptor.java:181)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.invokeInterceptor(DefaultFlushEntityEventListener.java:372)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.handleInterception(DefaultFlushEntityEventListener.java:349)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.scheduleUpdate(DefaultFlushEntityEventListener.java:287)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.onFlushEntity(DefaultFlushEntityEventListener.java:155)
    at org.hibernate.event.def.AbstractFlushingEventListener.flushEntities(AbstractFlushingEventListener.java:219)
    at org.hibernate.event.def.AbstractFlushingEventListener.flushEverythingToExecutions(AbstractFlushingEventListener.java:99)
    at org.hibernate.event.def.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:50)
    at org.hibernate.impl.SessionImpl.flush(SessionImpl.java:1208)
    at ome.logic.UpdateImpl.afterUpdate(UpdateImpl.java:294)
    at ome.logic.UpdateImpl.doAction(UpdateImpl.java:312)
    at ome.logic.UpdateImpl.doAction(UpdateImpl.java:302)
    at ome.logic.UpdateImpl.saveObject(UpdateImpl.java:108)
    at sun.reflect.GeneratedMethodAccessor655.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
    at ome.security.basic.EventHandler.invoke(EventHandler.java:154)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.orm.hibernate3.HibernateInterceptor.invoke(HibernateInterceptor.java:111)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:108)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at ome.tools.hibernate.ProxyCleanupFilter$Interceptor.invoke(ProxyCleanupFilter.java:241)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at ome.services.util.ServiceHandler.invoke(ServiceHandler.java:116)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
    at com.sun.proxy.$Proxy93.saveObject(Unknown Source)
    at sun.reflect.GeneratedMethodAccessor655.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
    at ome.security.basic.BasicSecurityWiring.invoke(BasicSecurityWiring.java:98)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at ome.services.blitz.fire.AopContextInitializer.invoke(AopContextInitializer.java:43)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
    at com.sun.proxy.$Proxy93.saveObject(Unknown Source)
    at sun.reflect.GeneratedMethodAccessor980.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at ome.services.blitz.util.IceMethodInvoker.invoke(IceMethodInvoker.java:179)
    at ome.services.throttling.Callback.run(Callback.java:56)
    at ome.services.throttling.InThreadThrottlingStrategy.callInvokerOnRawArgs(InThreadThrottlingStrategy.java:56)
    at ome.services.blitz.impl.AbstractAmdServant.callInvokerOnRawArgs(AbstractAmdServant.java:150)
    at ome.services.blitz.impl.UpdateI.saveObject_async(UpdateI.java:81)
    at sun.reflect.GeneratedMethodAccessor979.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
    at omero.cmd.CallContext.invoke(CallContext.java:59)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
    at com.sun.proxy.$Proxy94.saveObject_async(Unknown Source)
    at omero.api._IUpdateTie.saveObject_async(_IUpdateTie.java:122)
    at omero.api._IUpdateDisp.___saveObject(_IUpdateDisp.java:153)
    at omero.api._IUpdateDisp.__dispatch(_IUpdateDisp.java:389)
    at IceInternal.Incoming.invoke(Incoming.java:159)
    at Ice.ConnectionI.invokeAll(ConnectionI.java:2357)
    at Ice.ConnectionI.dispatch(ConnectionI.java:1208)
    at Ice.ConnectionI.message(ConnectionI.java:1163)
    at IceInternal.ThreadPool.run(ThreadPool.java:302)
    at IceInternal.ThreadPool.access$300(ThreadPool.java:12)
    at IceInternal.ThreadPool$EventHandlerThread.run(ThreadPool.java:643)
    at java.lang.Thread.run(Thread.java:722)

    serverExceptionClass = ome.conditions.OptimisticLockException
    message = You are not authorized to change the update event for ome.model.annotations.TagAnnotation:Id_2341 from ome.model.meta.Event:Id_68717 to ome.model.meta.Event:Id_68682
You may need to reload the object before continuing.
    backOff = 0
}
Traceback (most recent call last):
  File "./script", line 375, in <module>
    runScript()
  File "./script", line 363, in runScript
    message += processImages(client, scriptParams) + '\n'
  File "./script", line 217, in processImages
    channels, zselect, tselect, recalc, disableCdb)
  File "./script", line 166, in extractFeatures
    disableCdb)
  File "./script", line 86, in extractFeaturesOneChannel
    fids, features, ftset)
  File "/usr/lib/python2.6/site-packages/pyslid/database/direct.py", line 482, in update
    Answer = updateNameTag(conn, tag, DBfilename_new)
  File "/usr/lib/python2.6/site-packages/pyslid/database/direct.py", line 129, in updateNameTag
    conn.getUpdateService().saveObject(tag)
  File "/opt/omero/server/lib/python/omero/gateway/__init__.py", line 3460, in __call__
    return self.handle_exception(e, *args, **kwargs)
  File "/opt/omero/server/lib/python/omero/gateway/__init__.py", line 3457, in __call__
    return self.f(*args, **kwargs)
  File "/opt/omero/server/lib/python/omero_api_IUpdate_ice.py", line 107, in saveObject
    return _M_omero.api.IUpdate._op_saveObject.invoke(self, ((obj, ), _ctx))
omero.OptimisticLockException: exception ::omero::OptimisticLockException
{
    serverStackTrace = ome.conditions.OptimisticLockException: You are not authorized to change the update event for ome.model.annotations.TagAnnotation:Id_2341 from ome.model.meta.Event:Id_68717 to ome.model.meta.Event:Id_68682
You may need to reload the object before continuing.
    at ome.security.basic.OmeroInterceptor.managedEvent(OmeroInterceptor.java:1003)
    at ome.security.basic.OmeroInterceptor.checkManagedDetails(OmeroInterceptor.java:773)
    at ome.security.basic.OmeroInterceptor.resetDetails(OmeroInterceptor.java:316)
    at ome.security.basic.OmeroInterceptor.onFlushDirty(OmeroInterceptor.java:181)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.invokeInterceptor(DefaultFlushEntityEventListener.java:372)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.handleInterception(DefaultFlushEntityEventListener.java:349)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.scheduleUpdate(DefaultFlushEntityEventListener.java:287)
    at org.hibernate.event.def.DefaultFlushEntityEventListener.onFlushEntity(DefaultFlushEntityEventListener.java:155)
    at org.hibernate.event.def.AbstractFlushingEventListener.flushEntities(AbstractFlushingEventListener.java:219)
    at org.hibernate.event.def.AbstractFlushingEventListener.flushEverythingToExecutions(AbstractFlushingEventListener.java:99)
    at org.hibernate.event.def.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:50)
    at org.hibernate.impl.SessionImpl.flush(SessionImpl.java:1208)
    at ome.logic.UpdateImpl.afterUpdate(UpdateImpl.java:294)
    at ome.logic.UpdateImpl.doAction(UpdateImpl.java:312)
    at ome.logic.UpdateImpl.doAction(UpdateImpl.java:302)
    at ome.logic.UpdateImpl.saveObject(UpdateImpl.java:108)
    at sun.reflect.GeneratedMethodAccessor655.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
    at ome.security.basic.EventHandler.invoke(EventHandler.java:154)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.orm.hibernate3.HibernateInterceptor.invoke(HibernateInterceptor.java:111)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:108)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at ome.tools.hibernate.ProxyCleanupFilter$Interceptor.invoke(ProxyCleanupFilter.java:241)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at ome.services.util.ServiceHandler.invoke(ServiceHandler.java:116)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
    at com.sun.proxy.$Proxy93.saveObject(Unknown Source)
    at sun.reflect.GeneratedMethodAccessor655.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
    at ome.security.basic.BasicSecurityWiring.invoke(BasicSecurityWiring.java:98)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at ome.services.blitz.fire.AopContextInitializer.invoke(AopContextInitializer.java:43)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
    at com.sun.proxy.$Proxy93.saveObject(Unknown Source)
    at sun.reflect.GeneratedMethodAccessor980.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at ome.services.blitz.util.IceMethodInvoker.invoke(IceMethodInvoker.java:179)
    at ome.services.throttling.Callback.run(Callback.java:56)
    at ome.services.throttling.InThreadThrottlingStrategy.callInvokerOnRawArgs(InThreadThrottlingStrategy.java:56)
    at ome.services.blitz.impl.AbstractAmdServant.callInvokerOnRawArgs(AbstractAmdServant.java:150)
    at ome.services.blitz.impl.UpdateI.saveObject_async(UpdateI.java:81)
    at sun.reflect.GeneratedMethodAccessor979.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
    at omero.cmd.CallContext.invoke(CallContext.java:59)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
    at com.sun.proxy.$Proxy94.saveObject_async(Unknown Source)
    at omero.api._IUpdateTie.saveObject_async(_IUpdateTie.java:122)
    at omero.api._IUpdateDisp.___saveObject(_IUpdateDisp.java:153)
    at omero.api._IUpdateDisp.__dispatch(_IUpdateDisp.java:389)
    at IceInternal.Incoming.invoke(Incoming.java:159)
    at Ice.ConnectionI.invokeAll(ConnectionI.java:2357)
    at Ice.ConnectionI.dispatch(ConnectionI.java:1208)
    at Ice.ConnectionI.message(ConnectionI.java:1163)
    at IceInternal.ThreadPool.run(ThreadPool.java:302)
    at IceInternal.ThreadPool.access$300(ThreadPool.java:12)
    at IceInternal.ThreadPool$EventHandlerThread.run(ThreadPool.java:643)
    at java.lang.Thread.run(Thread.java:722)

    serverExceptionClass = ome.conditions.OptimisticLockException
    message = You are not authorized to change the update event for ome.model.annotations.TagAnnotation:Id_2341 from ome.model.meta.Event:Id_68717 to ome.model.meta.Event:Id_68682
You may need to reload the object before continuing.
    backOff = 0
}
}}}
```

--- a/scripts/Omero_Searcher_Feature_Calculation.py
+++ b/scripts/Omero_Searcher_Feature_Calculation.py
@@ -46,9 +46,14 @@ def extractFeaturesOneChannel(conn, image, scale, ftset, scaleSet,
     mid = 'image:%d c:%s z:%d t:%d' % (imageId, channels, zslice, timepoint)
 
     print 'Calculating features ftset:%s scale:%e %s' % (ftset, scale, mid)
-    [fids, features, scalec] = pyslid.features.calculate(
-        conn, imageId, scale, ftset, True, None,
-        pixels, channels, zslice, timepoint, debug=True)
+    try:
+        [fids, features, scalec] = pyslid.features.calculate(
+            conn, imageId, scale, ftset, True, None,
+            pixels, channels, zslice, timepoint, debug=True)
+    except pyslid.utilities.PyslidException as e:
+        m = 'Feature calculation failed for %s\nException:%s\n' % (mid, e)
+        sys.stderr.write(m)
+        return message + m
 
     if features is None:
         m = 'Feature calculation failed for %s\n' % mid

--- a/scripts/Omero_Searcher_Feature_Calculation.py
+++ b/scripts/Omero_Searcher_Feature_Calculation.py
@@ -308,7 +308,7 @@ def runScript():
         scripts.Long(
             'Select_Z', optional=False, grouping='5.1',
             description='Select Z index (starting from %d)' % IDX_OFFSET,
-            default=-1),
+            default=0),
 
 
         scripts.Bool(
@@ -319,7 +319,7 @@ def runScript():
         scripts.Long(
             'Select_T', optional=False, grouping='6.1',
             description='Select timepoint (starting from %d)' % IDX_OFFSET,
-            default=-1),
+            default=0),
 
 
         scripts.Bool(

--- a/scripts/Omero_Searcher_Feature_Calculation.py
+++ b/scripts/Omero_Searcher_Feature_Calculation.py
@@ -116,7 +116,7 @@ def extractFeatures(conn, image, scale, ftset, scaleSet,
     else:
         existing = listExistingCZTS(conn, imageId, ftset)
 
-    if zselect[0]:
+    if not zselect[0]:
         zslice = image.getSizeZ() / 2
     else:
         if (zselect[1] < IDX_OFFSET or
@@ -126,7 +126,7 @@ def extractFeatures(conn, image, scale, ftset, scaleSet,
             return message + m
         zslice = zselect[1] - IDX_OFFSET
 
-    if tselect[0]:
+    if not tselect[0]:
         timepoint = image.getSizeT() / 2
     else:
         if (tselect[1] < IDX_OFFSET or
@@ -185,8 +185,10 @@ def processImages(client, scriptParams):
                 scriptParams['Select_Readout_Channel'],
                 scriptParams['Select_Reference_Channel'])
 
-    zselect = (scriptParams['Use_Middle_Z'], scriptParams['Select_Z'])
-    tselect = (scriptParams['Use_Middle_T'], scriptParams['Select_T'])
+    zselect = (scriptParams['Select_Z_instead_of_middle'],
+               scriptParams['Select_Z'])
+    tselect = (scriptParams['Select_T_instead_of_middle'],
+               scriptParams['Select_T'])
 
     if scriptParams['Enable_Advanced_Options']:
         recalc = scriptParams['Recalculate_Existing_Features']
@@ -298,9 +300,10 @@ def runScript():
             default=IDX_OFFSET + 1),
 
 
-        scripts.Bool('Use_Middle_Z', optional=False, grouping='5',
-                       description='Which Z-slice to use',
-                       default=True),
+        scripts.Bool(
+            'Select_Z_instead_of_middle', optional=False, grouping='5',
+            description='Select a Z-slice instead of the default (middle)',
+            default=False),
 
         scripts.Long(
             'Select_Z', optional=False, grouping='5.1',
@@ -308,9 +311,10 @@ def runScript():
             default=-1),
 
 
-        scripts.Bool('Use_Middle_T', optional=False, grouping='6',
-                     description='Which timepoint to use',
-                     default=True),
+        scripts.Bool(
+            'Select_T_instead_of_middle', optional=False, grouping='6',
+            description='Select a timepoint instead of the default (middle)',
+            default=False),
 
         scripts.Long(
             'Select_T', optional=False, grouping='6.1',

